make-release.sh

```diff
@@ -0,0 +1,146 @@
+MAIN_BRANCH="main"
+TMP=""
+
+while [[ "$#" -gt 0 ]]; do	
+  case $1 in	
+    '-v'|'--version') VERSION="$2"; shift 1;;		
+    '-tmp'|'--use-tmp-dir') TMP=$(mktemp -d); shift 0;;
+  esac	
+  shift 1	
+done	
+
+bump_version () {	
+  CURRENT_BRANCH=$(git rev-parse --abbrev-ref HEAD)	
+
+  NEXT_VERSION=$1	
+  BUMP_BRANCH=$2	
+
+  git checkout "${BUMP_BRANCH}"	
+
+  echo "Updating project version to ${NEXT_VERSION}"	
+  echo "${VERSION}" > VERSION 
+  if [[ ${NOCOMMIT} -eq 0 ]]; then	
+    COMMIT_MSG="[release] Bump to ${NEXT_VERSION} in ${BUMP_BRANCH}"	
+    git commit -asm "${COMMIT_MSG}"	
+    git pull origin "${BUMP_BRANCH}"	
+
+    set +e
+    PUSH_TRY="$(git push origin "${BUMP_BRANCH}")"	
+    # shellcheck disable=SC2181	
+    set -e
+    if [[ $? -gt 0 ]] || [[ $PUSH_TRY == *"protected branch hook declined"* ]]; then	
+      PR_BRANCH=pr-${BUMP_BRANCH}-to-${NEXT_VERSION}	
+      # create pull request for the main branch branch, as branch is restricted	
+      git branch "${PR_BRANCH}"	
+      git checkout "${PR_BRANCH}"	
+      git pull origin "${PR_BRANCH}"	
+      git push origin "${PR_BRANCH}"	
+      lastCommitComment="$(git log -1 --pretty=%B)"	
+      hub pull-request -f -m "${lastCommitComment}" -b "${BUMP_BRANCH}" -h "${PR_BRANCH}"	
+    fi 	
+  fi	
+  git checkout "${CURRENT_BRANCH}"	
+}	
+
+usage ()	
+{	
+  echo "Usage: $0 --version [VERSION TO RELEASE]"	
+  echo "Example: $0 --version 0.1.0"; echo	
+}	
+
+if [[ ! ${VERSION} ]]; then	
+  usage	
+  exit 1	
+fi	
+
+
+# derive branch from version	
+BRANCH=${VERSION%.*}.x	
+
+# if doing a .0 release, use main branch; if doing a .z release, use $BRANCH	
+if [[ ${VERSION} == *".0" ]]; then	
+  BASEBRANCH="${MAIN_BRANCH}"	
+else 	
+  BASEBRANCH="${BRANCH}"	
+fi	
+
+# work in tmp dir
+if [[ $TMP ]] && [[ -d $TMP ]]; then
+  pushd "$TMP" > /dev/null || exit 1
+  # get sources from ${BASEBRANCH} branch
+  echo "Check out ${REPO} to ${TMP}/${REPO##*/}"
+  git clone "${REPO}" -q
+  cd "${REPO##*/}" || exit 1
+fi
+
+
+# get sources from ${BASEBRANCH} branch	
+git fetch origin "${BASEBRANCH}":"${BASEBRANCH}" || true
+git checkout "${BASEBRANCH}"	
+
+# create new branch off ${BASEBRANCH} (or check out latest commits if branch already exists), then push to origin
+if [[ "${BASEBRANCH}" != "${BRANCH}" ]]; then
+  git branch "${BRANCH}" || git checkout "${BRANCH}" && git pull origin "${BRANCH}" || true
+  git push origin "${BRANCH}"
+  git fetch origin "${BRANCH}:${BRANCH}" || true
+  git checkout "${BRANCH}"
+else
+  git fetch origin "${BRANCH}:${BRANCH}" || true
+  git checkout ${BRANCH}
+fi
+set -e
+
+# change VERSION file	
+echo "${VERSION}" > VERSION
+
+QUAY_REPO="quay.io/devfile/devworkspace-controller:${VERSION}"
+docker build -t "${QUAY_REPO}" -f ./build/Dockerfile .
+docker push "${QUAY_REPO}"
+
+# replace image version in default environment
+sed -i "s/IMG=quay.io\/devfile\/devworkspace-controller:.*/IMG=quay.io\/devfile\/devworkspace-controller:${VERSION}/" ./deploy/generate-deployment.sh
```

Comment:
I think `./deploy/generate-deployment.sh --use-defaults` should be adapted to make the work without sed

Reply:
sure, later we could probably make it so that default env variables could be overriden by user, if it wouldn't break anything

Reply:
This would also commit the modified `generate-deployment.sh` file, is that something we want to do? I'm fine with this for now but agree we should maybe make `generate-deployment.sh` take a `--tag` parameter or similar.